pkg/syncer/specsyncer.go

```diff
@@ -137,26 +140,16 @@ func (c *Controller) upsertIntoDownstream(ctx context.Context, gvr schema.GroupV
 	// TODO: wipe things like finalizers, owner-refs and any other life-cycle fields. The life-cycle
 	//       should exclusively owned by the syncer. Let's not some Kubernetes magic interfere with it.
 
-	if _, err := c.toClient.Resource(gvr).Namespace(namespace).Create(ctx, obj, metav1.CreateOptions{}); err != nil {
-		if !k8serrors.IsAlreadyExists(err) {
-			klog.Errorf("Creating resource %s/%s: %v", namespace, obj.GetName(), err)
-			return err
-		}
-
-		existing, err := c.toClient.Resource(gvr).Namespace(namespace).Get(ctx, obj.GetName(), metav1.GetOptions{})
-		if err != nil {
-			klog.Errorf("Getting resource %s/%s: %v", namespace, obj.GetName(), err)
-			return err
-		}
-		klog.Infof("Object %s/%s already exists: update it", gvr.Resource, obj.GetName())
+	data, err := json.Marshal(obj)
+	if err != nil {
+		return err
+	}
 
-		obj.SetResourceVersion(existing.GetResourceVersion())
-		if _, err := c.toClient.Resource(gvr).Namespace(namespace).Update(ctx, obj, metav1.UpdateOptions{}); err != nil {
-			klog.Errorf("Updating resource %s/%s: %v", namespace, obj.GetName(), err)
-			return err
-		}
-		return nil
+	if _, err := c.toClient.Resource(gvr).Namespace(namespace).Patch(ctx, obj.GetName(), types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: syncerApplyManager, Force: pointer.Bool(true)}); err != nil {
+		klog.Errorf("Applying resource %s/%s: %v", namespace, obj.GetName(), err)
```

Comment:
nit: we should expose the metadata.clusterName for clarity here

Reply:
`obj.GetClusterName()` is now added to the logging statements.